### PR TITLE
Do not check builds/details in build by strategy admission control

### PR DESCRIPTION
**Pull request by csrwng**
Fixes BZ 1298861

This was a regression introduced by #6576.

### CODE REVIEW

**Comment by csrwng** [comment]
@bparees PTAL

[test]
[testonlyextended][extended:core(builds)]

File: pkg/build/admission/admission.go [hunk 1]
```diff
@@ -49,6 +49,11 @@ func (a *buildByStrategy) Admit(attr admission.Attributes) error {
 	if resource := attr.GetResource(); resource != buildsResource && resource != buildConfigsResource {
 		return nil
 	}
+	// Explicitly exclude the builds/details subresource because it's only
+	// updating commit info and cannot change build type.
+	if attr.GetResource() == "builds" && attr.GetSubresource() == "details" {
```
Comment (bparees) [inline pkg/build/admission/admission.go]: s/"builds"/buildsResource"

**Comment by bparees** [comment]
one nit and lgtm.  over to @smarterclayton

**Comment by bparees** [comment]
also fyi test+testonlyextended doesn't make much sense :)

**Comment by csrwng** [comment]
updated.

Should the tag be "testextended" ?

**Comment by smarterclayton** [comment]
Approved, merge when ready

**Comment by bparees** [comment]
"[test][extended:core(builds)]"  is what you wanted (but it's also what you got as a result of what you did..  just saying the onlyextended does not actually limit it to only extended tests if you also put a test tag on it)

**Comment by csrwng** [comment]
[test][extended:core(builds)]

**Comment by liggitt** [comment]
@smarterclayton for approval to prevent regression in builders updating build details

**Comment by liggitt** [comment]
nm, saw comment

**Comment by bparees** [comment]
@csrwng those extended test failures look scary, can you take a look?

**Comment by bparees** [comment]
@csrwng this one: https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin/8636/consoleFull

going to run it again in the meantime after cleaning up the test tags.

[test]

**Comment by csrwng** [comment]
@bparees after the failure below every test fails. It looks like the server stopped responding after that point. I didn't see this when I ran extended tests earlier today, so I think something must have gone wrong with the vm. 
```

  Expected error:
      <*url.Error | 0xc2087a6180>: {
          Op: "Get",
          URL: "https://172.18.2.235:8443/api/v1/namespaces/extended-test-build-image-source-dx2lb/endpoints?timeout=30s",
          Err: {
              Op: "dial",
              Net: "tcp",
              Addr: {
                  IP: "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xff\xff\xac\x12\x02\xeb",
                  Port: 8443,
                  Zone: "",
              },
              Err: 0x6f,
          },
      }
      Get https://172.18.2.235:8443/api/v1/namespaces/extended-test-build-image-source-dx2lb/endpoints?timeout=30s: dial tcp 172.18.2.235:8443: connection refused
  not to have occurred
```

**Comment by bparees** [comment]
ok we'll see what the current run yields.

**Comment by smarterclayton** [comment]
[test]

**Comment by csrwng** [comment]
@smarterclayton extended passed in one of the tests: 
https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin/8650/consoleFull

and failed with the same exact error above in the other test:
```
  Expected error:
      <*url.Error | 0xc2087a6180>: {
          Op: "Get",
          URL: "https://172.18.2.235:8443/api/v1/namespaces/extended-test-build-image-source-dx2lb/endpoints?timeout=30s",
          Err: {
              Op: "dial",
              Net: "tcp",
              Addr: {
                  IP: "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xff\xff\xac\x12\x02\xeb",
                  Port: 8443,
                  Zone: "",
              },
              Err: 0x6f,
          },
      }
      Get https://172.18.2.235:8443/api/v1/namespaces/extended-test-build-image-source-dx2lb/endpoints?timeout=30s: dial tcp 172.18.2.235:8443: connection refused
  not to have occurred
```

**Comment by csrwng** [comment]
I don't think that error is related to this pull

**Comment by csrwng** [comment]
[test]

**Comment by csrwng** [comment]
I wanted to kick off regular tests without the extended tests and then extended tests separately, but it doesn't look like it's letting me do that. It kicked off both again.

**Comment by csrwng** [comment]
Started 2 sets of tests manually... 
Regular test: https://ci.openshift.redhat.com/jenkins/job/test_pr_origin/54/
Extended only: https://ci.openshift.redhat.com/jenkins/job/test_pr_origin/55/

**Comment by bparees** [comment]
It aggregates all the tags. If you want extended separate you can add the
testonlyextended tag again (Dan educated me this evening) and you'll get a
separate run for that.

Ben Parees | OpenShift

**Comment by bparees** [comment]
1 for 2 on the test job, 2 for 2 on extended.  test failure looks unrelated.
[merge]

**Pull request merged**